### PR TITLE
don't install nextcloud/ocp for workflows that don't require it

### PR DESCRIPTION
**Pull request by icewind1991**
Add don't install the version of `nextcloud/ocp` specified in `composer.json` for psalm when we are just going to install a different version after it anyway.

This makes it a bit easier for apps to support a larger range of server versions.

### CODE REVIEW

File: workflow-templates/lint-php-cs.yml [hunk 1]
```diff
@@ -39,7 +39,9 @@ jobs:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
       - name: Install dependencies
-        run: composer i
+        run: |
+          composer remove nextcloud/ocp --dev
```
Comment (susnux) [inline workflow-templates/lint-php-cs.yml:43]: We do not install the server here, so I think we need it.

File: workflow-templates/openapi.yml [hunk 2]
```diff
@@ -64,7 +64,9 @@ jobs:
           npm ci
 
       - name: Set up dependencies
-        run: composer i
+        run: |
+          composer remove nextcloud/ocp --dev
```
Comment (susnux) [inline workflow-templates/openapi.yml:68]: In this workflow also server is not cloned, so no OCP available.

Comment (icewind1991) [reply workflow-templates/openapi.yml]: removed the change

**Review by susnux — APPROVED** [review]

File: workflow-templates/lint-php-cs.yml [hunk 3]
```diff
@@ -42,7 +42,9 @@ jobs:
...
+          composer i
```
Comment (nickvergessen) [inline workflow-templates/lint-php-cs.yml:47]: shouldn't
```suggestion
          composer i --no-dev
```

be enough for code style?

Comment (susnux) [reply workflow-templates/lint-php-cs.yml:47]: `php-cs-fixer` not a development dependency?

Comment (nickvergessen) [reply workflow-templates/lint-php-cs.yml:47]: Arg :speak_no_evil:

**Review by skjnldsv — APPROVED** [review]

**Comment by skjnldsv** [comment]
Conflicts!

**Comment by icewind1991** [comment]
resolved

**Review by susnux — APPROVED** [review]

**Pull request merged**